migration-engine/connectors/sql-migration-connector/src/sql_renderer.rs

```diff
@@ -28,4 +26,20 @@ pub(crate) trait SqlRenderer {
     fn render_references(&self, schema_name: &str, foreign_key: &ForeignKey) -> String;
 
     fn render_default<'a>(&self, default: &'a DefaultValue, family: &ColumnTypeFamily) -> Cow<'a, str>;
+
+    /// Attempt to render a database-specific ALTER COLUMN based on the
+    /// passed-in differ. `None` means that we could not generate a good (set
+    /// of) ALTER COLUMN(s), and we should fall back to dropping and recreating
+    /// the column.
+    ///
+    /// The return type should be interpreted as (alter table lines, (statement before, statement after))
+    fn render_alter_column(&self, differ: &ColumnDiffer<'_>) -> Option<RenderedAlterColumn>;
+}
+
+#[derive(Default)]
+pub(crate) struct RenderedAlterColumn {
+    /// The statements that will be included in the ALTER TABLE
+    pub(crate) alter_columns: Vec<String>,
+    /// The statements to be run before and after the ALTER TABLE.
+    pub(crate) before_and_after: Option<(String, String)>,
```

Comment:
I don't like this. I think in the short to medium term, `ALTER TABLE` should be backend-specific, with shared util functions for what is database agnostic, rather than the reverse, then the `RenderedAlterColumn` can go.

Reply:
I didn't do it in this PR because it's already large enough, and I didn't want to cause even more churn.